### PR TITLE
OGC Processes API v2

### PR DESCRIPTION
**Pull request by jkeifer**
See list of commits for a full changelog.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,32 +1,24 @@
...
-    SWOOP_S3_ENDPOINT=$SWOOP_S3_ENDPOINT  \
-    SWOOP_BUCKET_NAME=$SWOOP_BUCKET_NAME  \
-    SWOOP_EXECUTION_DIR=$SWOOP_EXECUTION_DIR  \
-    SWOOP_WORKFLOW_CONFIG_FILE=$SWOOP_WORKFLOW_CONFIG_FILE \
-    PGDATABASE=$PGDATABASE \
-    PGHOST=$PGHOST \
-    PGUSER=$PGUSER
+ENV SWOOP_CONFIG_FILE=swoop-config.yml
+
+RUN env
```
Comment (jkeifer) [inline Dockerfile:16]: remove

File: src/swoop/api/io.py [hunk 2]
```diff
@@ -2,22 +2,48 @@
...
+        raise ValueError(f"invalid endpoint: '{s3_endpoint}'")
+
+    return secure, s3_endpoint
+
+
 class IOClient:
-    def __init__(
-        self, s3_endpoint: str, access_key: str, secret_key: str, bucket_name: str
-    ):
+    def __init__(self, bucket_name: str, s3_endpoint: str = "s3.amazonaws.com"):
```
Comment (jkeifer) [inline src/swoop/api/io.py:31]: review this more closely, looks like the split doesn't work right

File: src/swoop/api/io.py [hunk 3]
```diff
@@ -64,21 +90,18 @@ def delete_object(self, object_name: str):
...
-    def delete_bucket(self, bucket_name=None):
-        self.bucket_name = bucket_name if bucket_name else self.bucket_name
-        if self.bucket_name and self.client.bucket_exists(self.bucket_name):
+    def delete_bucket(self):
+        if self.client.bucket_exists(self.bucket_name):
             self.client.remove_bucket(self.bucket_name)
             logger.debug(f"deleted bucket: {self.bucket_name}")
-            self.bucket_name = None
 
     def bucket_exists(self, bucket_name: str):
```
Comment (jkeifer) [inline src/swoop/api/io.py:103]: bucket name still?

File: src/swoop/api/models/shared.py [hunk 4]
```diff
@@ -99,127 +69,70 @@ class Metadata(BaseModel):
...
 
-class InlineOrRefData(BaseModel):
-    __root__: InputValueNoObject | QualifiedInputValue | Link
+InlineOrRefData = RootModel[InputValueNoObject | QualifiedInputValue | Link]
 
 
-class Results(BaseModel):
-    # TODO: This becomes the schema of the workflow output
-    __root__: dict[str, InlineOrRefData] | None = None
+# TODO: This becomes the schema of the workflow output
```
Comment (jkeifer) [inline src/swoop/api/models/shared.py:115]: Addressed?

File: src/swoop/api/routers/jobs.py [hunk 5]
```diff
@@ -291,11 +290,83 @@ async def get_workflow_execution_result(
     results = request.app.state.io.get_object(f"/executions/{jobID}/output.json")
```
Comment (jkeifer) [inline src/swoop/api/routers/jobs.py:291]: Add TODO for missing error behavior

**Review by hectormachin — APPROVED** [review]

File: src/swoop/api/routers/jobs.py (hunk 5 above)
Comment (jkeifer) [reply src/swoop/api/routers/jobs.py:329]: Fixed this by extracting the check into a function we can reuse here.

**Pull request merged**